agent/battle/vision/trigger.py

```
@@ -0,0 +1,123 @@
+        if context.current_scene is Scene.UNKNOWN:
+            if context.elapsed_since_action_ms < self.transition_guard_ms:
+                return TriggerDecision(False, ("transition_guard",), ())
+            if not context.frame_stable:
+                return TriggerDecision(False, ("frame_unstable",), ())
+            if context.consecutive_unknown < self.min_unknown_samples:
+                return TriggerDecision(False, ("unknown_not_persistent",), ())
+            return TriggerDecision(True, ("stable_unknown_scene",), ("scene", "dialogs"))
+
+        requested = self._missing_fields(context)
+        if requested:
+            return TriggerDecision(True, ("requested_fields_missing",), requested)
```

Ruff (0.16.1)

[warning] Boolean positional value in function call

(FBT003)